src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/VisualStudioWindowsProjectCapabilityResolver.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    [Export(typeof(ProjectCapabilityResolver))]
+    internal class VisualStudioWindowsProjectCapabilityResolver : ProjectCapabilityResolver
+    {
+        public override bool HasCapability(string documentFilePath, object project, string capability)
+        {
+            if (project is not IVsHierarchy vsHierarchy)
+            {
+                return false;
+            }
+
+            var localHasCapability = LocalHasCapability(vsHierarchy, capability);
+            return localHasCapability;
+        }
+
+        private static bool LocalHasCapability(IVsHierarchy hierarchy, string capability)
+        {
+            if (hierarchy is null)
+            {
+                return false;
+            }
+
+            try
+            {
+                var hasCapability = hierarchy.IsCapabilityMatch(capability);
+                return hasCapability;
+            }
+            catch (NotSupportedException)
+            {
+                // IsCapabilityMatch throws a NotSupportedException if it can't create a
+                // BooleanSymbolExpressionEvaluator COM object
```

Comment:
Why might it not be able to do that?

Reply:
I don't fully recall to be honest. This code was originally from the `DefaultProjectHierarchyInspector`: https://github.com/dotnet/razor-tooling/pull/6139/files#diff-95ba9af8d6a0ae220cfbf561e49a5f7d567a17edeea7c74f5a2ba4c40dd4707bL68